sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1188,14 +1188,14 @@ private String getLinkName(String sessionId) {
     }
 
     /**
-     * Map the error to {@link ServiceBusAmqpException}
+     * Map the error to {@link ServiceBusReceiverException}
      */
     private Throwable mapError(Throwable throwable, ServiceBusErrorSource errorSource) {
-        if ((throwable instanceof ServiceBusAmqpException) || !(throwable instanceof AmqpException)) {
-            return throwable;
-        } else {
-            return new ServiceBusAmqpException((AmqpException) throwable, errorSource);
+        // If it is already `ServiceBusReceiverException`, we can just throw it.
+        if (!(throwable instanceof AmqpException) && !(throwable instanceof ServiceBusReceiverException)) {
```

Comment:
Why are we not wrapping AmqpException?

Reply:
based on our discussion converting all the exception to SBRE .